packages/google_maps_flutter/google_maps_flutter_android/lib/src/google_maps_flutter_android.dart

```diff
@@ -1166,6 +1165,34 @@ PlatformJointType platformJointTypeFromJointType(JointType jointType) {
   return PlatformJointType.mitered;
 }
 
+/// Converts a PatternItem to Pigeon's PlatformPatternItem for PlatformPolyline
+/// pattern member.
+@visibleForTesting
+PlatformPatternItem platformPatternItemFromPatternItem(PatternItem item) {
+  final List<Object> json = item.toJson() as List<Object>;
+  final String tag = json[0] as String;
+  PlatformPatternItemType type;
+  double? length;
+
+  /// These string values identify the type of pattern. They are defined and
+  /// used in the PatternItem class's factory methods in
+  /// lib/src/types/pattern_item.dart, in the
+  /// google_maps_flutter_platform_interface package.
```

Comment:
This is basically the same problem as camera update; the real solution is to have structured information (either subclasses like camera update, or maybe just a type enum and an optional field for length since it seems plausible that those would be enough for the future).

You don't need to fix that here since this instance isn't as egregiously problematic as camera update was, but please file an issue to track it and reference it in a TODO here so it's clear to future us that this isn't desirable code.